neopixel.py

```diff
@@ -199,6 +199,30 @@ def set_pixel(self, pixel_num, rgb_w, how_bright=None):
         else:
             self.pixels[pixel_num] = pix_value
 
+    def get_pixelRGB(self, pixel_num):
+        """
+        Get red, green and blue value of pixel on position <pixel_num>
+
+        :param pixel_num: Index of pixel to be set or slice object representing multiple leds
+        :return rgb_w: Tuple of form (r, g, b) or (r, g, b, w) representing color to be used
+        """
+        balance = self.pixels[pixel_num]
+        if self.W_in_mode:
+            w = balance & 255
+            balance = (balance-w) >> 8
+        b = balance & 255
+        balance = (balance-b) >> 8
+        r = balance & 255
+        g = ((balance-r) >> 8) & 255
```

Comment:
This won't work for every configuration. I'd recommend that this is changed to:
```suggestion
        if self.W_in_mode:
            w = (balance >> sh_W) & 255

        b = (balance >> sh_B) & 255
        r = (balance >> sh_R) & 255
        g = (balance >> sh_G) & 255
```
You can verify that all the values are the same with following code:
```python
from itertools import permutations
perm_rgb = []
for perm in permutations("RGBW", 4):
    strip = Neopixel(num_leds, "".join(perm))
    strip.brightness(25.5)
    strip[0] = (10, 20, 30, 40)
    perm_rgb.append(strip.get_pixelRGB(0))

print("Should be True: ", all(el == perm_rgb[0] for el in perm_rgb))```

Reply:
Made the suggested change. I couldn't test with permuations as I could not find a micropython implementation of itertools which supported it. But I have tested with some physical hardware (an Adafruit RGBW neopixel ring) which I bought, and confirmed I could set and read back colours from these RGBW LEDs.